hw/Makefile

```diff
@@ -97,11 +97,14 @@ blk-hjson      = $(PRJ_DIR)/hw/ip/$*/$(dir_hjson)/$*.hjson
 # If the ip has a local generation script, run it first. This will
 # require all blocks to use a consistent naming. This should be hooked
 # into ipgen (see #5636) for completeness.
-$(ips_reg): %_reg: otp-mmap lc-state-enc | $(REG_OUTPUT_DV_DIR)
+$(ips_reg): %_reg: lc-state-enc | $(REG_OUTPUT_DV_DIR)
 	$(if $(wildcard $(blk-gen-script)),$(blk-gen-script))
 	${PRJ_DIR}/util/regtool.py ${toolflags} -r $(blk-hjson)
 	${PRJ_DIR}/util/regtool.py -s -t $(REG_OUTPUT_DV_DIR) $(blk-hjson)
 
+# Register generation for otp_ctrl also depends on running gen-otp-mmap.py
+.PHONY: otp-mmap
```

Comment:
is it generally good practice to declare `PHONY` where its needed and not slap them into one line?
this does makes sense to me..but wanted to double check.